engine/core/src/main/java/org/datacleaner/job/builder/AnalysisJobBuilder.java

```diff
@@ -310,11 +310,23 @@ public void importJob(AnalysisJob job) {
     }
 
     public AnalysisJobBuilder removeSourceColumn(MetaModelInputColumn inputColumn) {
-        final boolean removed = _sourceColumns.remove(inputColumn);
-        if (removed) {
+        final int index = _sourceColumns.indexOf(inputColumn);
+        if (index != -1) {
+            final MetaModelInputColumn removedColumn = _sourceColumns.remove(index);
```

Comment:
Using index we get the exact same input column, not just one that is equal. This turned out to be important or else some property values would not update when calling ComponentBuilder.removeInputColumn(...)